Cargo.toml

```diff
@@ -16,6 +16,7 @@ categories = ["network programming"]
 async-trait = "0.1.64"
 delay_map = "0.3.0"
 futures = "0.3.26"
+parking_lot = "0.11.2"
```

Comment:
I'm 👎🏻 making a performance-driven change without benchmarks.

Here's a thread discussing the tradeoffs, and it's far from an obvious win: https://users.rust-lang.org/t/which-mutex-to-use-parking-lot-or-std-sync/85060/5

I'd say let's simplify this PR by pulling it back out. If we want to grind on performance in uTP, we can build more benchmarks and experiment. But I don't think uTP is our core performance issue (you alluded to the same thing in the call, based on your profiling results).

Reply:
I removed it. All RwLocks in Trin use parking_lot maybe that is something we should look into then?